extmod/vfs_fat_file.c

```diff
@@ -140,6 +140,11 @@ STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(file_obj___exit___obj, 4, 4, file_obj
 STATIC mp_uint_t file_obj_ioctl(mp_obj_t o_in, mp_uint_t request, uintptr_t arg, int *errcode) {
     pyb_file_obj_t *self = MP_OBJ_TO_PTR(o_in);
 
+    if (request == MP_STREAM_FLUSH) {
+	file_obj_flush(o_in);
+	return 0;
+    }
```

Comment:
It would be better to inline (and simplify) file_obj_flush here, and then remove file_obj_flush and replace file_obj_flush_obj with mp_stream_flush_obj.  That would follow how the unix port does it.

Also, please follow the if-logic here and combine the if with the below if-else chain.